test/broadcast.jl

```diff
@@ -360,7 +360,7 @@ end
 let f17314 = x -> x < 0 ? false : x
     @test eltype(broadcast(f17314, 1:3)) === Int
     @test eltype(broadcast(f17314, -1:1)) === Integer
-    @test eltype(broadcast(f17314, Int[])) == Union{Bool,Int}
+    @test eltype(broadcast(f17314, Int[])) === Integer
```

Comment:
This is a *minor change* which I think can be considered as a bug fix, in the sense that before this PR the element type when the input is empty will never be observed when the array isn't empty (we can only ever observe `Int`, `Bool` or `Integer`). I could change the PR to preserve the existing behavior if we want (e.g. for backports).

Reply:
Oh that's interesting.  Yeah, here's the current behaviors:

```
julia> eltype(broadcast(f17314, Int[]))
Union{Bool, Int64}

julia> eltype(broadcast(f17314, Int[1]))
Int64

julia> eltype(broadcast(f17314, Int[-1]))
Bool

julia> eltype(broadcast(f17314, Int[1,-1]))
Integer
```

The reason for using inference here in the first place is to preserve the performance in the non-empty case.  Adding a _fourth_ possible return type defeats such a purpose, so I'm in support of this change.

Reply:
Exactly - we either need to change the last to match the first, or the first to match the last.

This PR seems the least breaking, and suitable for v1.x. If we ever wanted to consider the other way around maybe that should be a v2.0 change?